source/simulator/checkpoint_restart.cc

```diff
@@ -113,15 +114,14 @@ namespace aspect
 
       //If we are using a free surface, also serialize the mesh vertices vector, which
       //uses its own dof handler
-      std::vector<const LinearAlgebra::Vector *> x_fs_system (2);
+      std::vector<const LinearAlgebra::Vector *> x_fs_system (1);
       std_cxx11::unique_ptr<parallel::distributed::SolutionTransfer<dim,LinearAlgebra::Vector> > freesurface_trans;
       if (parameters.free_surface_enabled)
         {
           freesurface_trans.reset (new parallel::distributed::SolutionTransfer<dim,LinearAlgebra::Vector>
                                    (free_surface->free_surface_dof_handler));
 
-          x_fs_system[0] = &free_surface->mesh_vertices;
-          x_fs_system[1] = &free_surface->mesh_vertex_velocity;
+          x_fs_system[0] = &free_surface->mesh_displacements;
```

Comment:
This change is ok, but if you felt challenged, I think the "right" way to do it would be if the free surface handler attached itself to some signals that are triggered right where this block is located, and where it all is unpacked below. This way, we could move some more code out of the core and into the free surface handler itself.